app/main/views/services.py

```diff
@@ -79,6 +83,48 @@ def list_services(framework_slug):
     ), 200
 
 
+@main.route("/frameworks/<string:framework_slug>/all-services")
+@login_required
+def list_all_services(framework_slug):
+    if not (is_g12_recovery_supplier(current_user.supplier_id) and framework_slug == "g-cloud-12"):
+        abort(404)
+
+    framework = get_framework_or_404(data_api_client, framework_slug, allowed_statuses=['live'])
+
+    suppliers_services = data_api_client.find_services(
+        supplier_id=current_user.supplier_id,
+        framework=framework_slug,
+    )["services"]
```

Comment:
Interesting question; probably the same thing that happens here 😁 

https://github.com/alphagov/digitalmarketplace-supplier-frontend/blob/master/app/main/views/services.py#L65

Reply:
OK. This is controlled in the API by `DM_API_SERVICES_PAGE_SIZE`, which is set to `100`. So I think this is a real bug that already exists in `list_services` - if a supplier has more than 100 services, they will not all be listed.

I don't know if this is likely to affect any suppliers - do we know the maximum number of services for one supplier?

In any case, this is a pre-existing bug, so not the end of the world if left as-is. I'll raise a quick PR to fix in `list_services`.

Reply:
> do we know the maximum number of services for one supplier?

There is no maximum on our side; anecdotally I think there are suppliers with over 400 services.

Reply:
Experimentally though it seems like `find_services` can return more than 100 services.

Reply:
Agree - I can see that a supplier with >>100 services can see all their services in the list on staging. So there's actually no bug - though I still don't understand how...

Reply:
OK. I now understand what's going on. The result from `find_services` is only paginated if you don't provide a supplier ID.

We do provide a supplier ID, so we get the full list of services.

So there is no bug, but this could be much better documented...

Reply:
https://github.com/alphagov/digitalmarketplace-apiclient/pull/240